radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/bft/VerifiedVertexStoreState.java

```diff
@@ -167,28 +166,30 @@ public static VerifiedVertexStoreState create(
           String.format("highQC=%s highQC proposed does not have a corresponding vertex", highQC));
     }
 
-    return new VerifiedVertexStoreState(highQC, rootHeader, root, idToVertex, vertices, highestTC);
+    return new VerifiedVertexStoreState(
+        highQC, headers.getSecond(), root, ImmutableMap.copyOf(seen), vertices, highestTC);
   }
 
   public VerifiedVertexStoreState prune(Hasher hasher) {
-    if (highQC.highestQC().getCommittedAndLedgerStateProof(hasher).isPresent()) {
-      Pair<BFTHeader, LedgerProof> newHeaders =
-          highQC.highestQC().getCommittedAndLedgerStateProof(hasher).get();
-      BFTHeader header = newHeaders.getFirst();
+    var stateProof = highQC.highestQC().getCommittedAndLedgerStateProof(hasher);
+    if (stateProof.isPresent()) {
+      var newHeaders = stateProof.get();
+      var header = newHeaders.getFirst();
+
       if (header.getView().gt(root.getView())) {
-        VerifiedVertex newRoot = idToVertex.get(header.getVertexId());
-        ImmutableList<VerifiedVertex> newVertices =
+        var newRoot = idToVertex.get(header.getVertexId());
+        var newVertices =
             ImmutableList.of(
                 idToVertex.get(highQC.highestQC().getParent().getVertexId()),
                 idToVertex.get(highQC.highestQC().getProposed().getVertexId()));
-        ImmutableMap<HashCode, VerifiedVertex> idToVertex =
+        var idToVertexMap =
```

Comment:
do we need to add "Map" to the name?

Reply:
The original name shadowed the other variable, and it was reported by Sonar as code smell.